src/components/client.tsx

```diff
@@ -58,9 +76,30 @@ export class UrqlClient<MutationDeclarations> extends Component<
     if (JSON.stringify(prevProps.query) !== JSON.stringify(this.props.query)) {
       this.state.client.executeQuery(this.props.query);
     }
+
+    if (
+      JSON.stringify(prevProps.subscriptions) !==
+      JSON.stringify(this.props.subscriptions)
+    ) {
+      this.deactiveSubscriptions(prevProps.subscriptions);
+      this.activateSubscriptions();
+    }
   }
 
+  deactiveSubscriptions = (subscriptions: Subscription[] = []) => {
+    subscriptions.forEach(sub =>
+      this.state.client.executeUnsubscribeSubscription(sub)
+    );
+  };
+
+  activateSubscriptions = () => {
+    (this.props.subscriptions || []).forEach(sub =>
+      this.state.client.executeSubscription(sub)
```

Comment:
I find it weird that every single subscription is handled separately here. I think the more idiomatic Rx approach would be to merge them all here in the component into a single subscription / observable